src/plugins/auto/src/infer_request.cpp

```diff
@@ -56,10 +56,18 @@ ov::auto_plugin::InferRequest::InferRequest(const std::shared_ptr<const ov::auto
         }
     } else {
         for (const auto& input : get_inputs()) {
-            ov::ISyncInferRequest::set_tensor(input, m_shared_request->get_tensor(input));
+            auto tensor = m_shared_request->get_tensor(input);
+            if (!tensor._so) {
+                tensor._so = m_shared_request._so;
+            }
+            ov::ISyncInferRequest::set_tensor(input, tensor);
         }
         for (const auto& output : get_outputs()) {
-            ov::ISyncInferRequest::set_tensor(output, m_shared_request->get_tensor(output));
+            auto tensor = m_shared_request->get_tensor(output);
+            if (!tensor._so) {
+                tensor._so = m_shared_request._so;
+            }
+            ov::ISyncInferRequest::set_tensor(output, tensor);
```

Comment:
why our Lifetime tests don't catch it?

Reply:
1.0 plugin works fine together with wrapper. when plugin goes to 2.0(e.g GPU), life time tests catches this failure.

Reply:
@songbell Shouldn't we override get_tensor() default implementations too?

Reply:
set_tensor call here already set the tensor with so to m_tensors in base class, get_tensor() will return the correct tensor with so, I don't see the need to override get_tensor() here, is my understanding correct?

Reply:
Ok, in this case, looks like it will work. 
Did you check this change on GPU 2.0 branch?

Reply:
checked on GPU2.0 branch, but found batch failure, updated the PR